odoo/addons/test_lint/tests/test_manifests.py

```diff
@@ -124,3 +133,53 @@ def _test_manifest_countries_value(self, module, values):
                     "Country value %s specified for the icon in manifest of module %s doesn't look like a country code"
                     "Please specify a correct value or remove this key from the manifest.",
                     value, module)
+
+    @staticmethod
+    def _get_data_filenames_from_manifest(manifest_data):
+        """Get data file names from the manifest keys"""
+        data_filenames = set()
+        for data_key in DFTL_MANIFEST_DATA_KEYS:
+            for file_name in manifest_data.get(data_key) or []:
+                if splitext(file_name)[1].lower() not in MANIFEST_DATA_EXTS:
+                    continue
+                data_filenames.add(file_name)
+        return data_filenames
+
+    @staticmethod
+    def _get_data_filenames_from_module(module):
+        """Get data file names from the module path"""
+        module_root_path_obj = Path(file_path(module)).resolve()
+        fnames = set()
```

Comment:
Instead of using a set (which has random ordering), you could just make this function a generator.